### DISCUSSION TITLE
What is the purpose of vcpkg?

### DISCUSSION THREAD

**Discussion by cdclawson** [microsoft/vcpkg#18875] [Q&A] [answered]
I'm confused about what it's for. Is it intended to produce stand-alone Windows binaries? Is it a shell/programming environment like WSL or Cygwin? Is it for making binaries or libraries that integrate into other projects? Something else? Thanks

**Comment by mathisloge** [comment, marked as answer]
vcpkg is a package manager for C/C++ libraries. This means that libraries can be built in a simplified way (without having to deal much with the build specifics of each library). This is a special circumstance of c /c++ libraries, because there are no unified building tools. Each library has mostly so its peculiarity, how it is to be built. This is not particularly nice, but has arisen through the long history of C/C++.
Now if you need e.g. boost for your program/library, you have three ways: build from source, use the provided installer for your platform or use a package manager like vcpkg. 

So your statement 
> Is it for making binaries or libraries that integrate into other projects?

is somewhat correct.  vcpkg contains the build instructions to build a project and can instruct your compiler to follow these build instructions and build the library. These build instructions can be located at https://github.com/microsoft/vcpkg/tree/master/ports

Hope this clarifies vcpkg a bit.

**Reply by cdclawson** [reply]
Thank you.

**Comment by skycommand** [comment]
Perhaps an example can explain things best.

For me, VCpkg is a hassle-free way of building FFmpeg. The resulting FFmpeg isn't perfect, but it does the job, and VCpkg is so error-resilient and resume-friendly. The alternative way of building FFmpeg would be through `m-ab-s/media-autobuild_suite`, which is ... not hassle-free.

**Reply by cdclawson** [reply]
Thank you.
